src/libbson/src/bson/bson.c

```diff
@@ -143,14 +175,20 @@ _bson_impl_alloc_grow (bson_impl_alloc_t *impl, /* IN */
    /*
     * Determine how many bytes we need for this document in the buffer
     * including necessary trailing bytes for parent documents.
+    *
+    * Note that the buffer offset and nesting depth are not available
+    * outside bson_impl_alloc_t, meaning it's not possible for callers to
+    * fully rule out BSON_MAX_SIZE overflow before _bson_grow().
```

Comment:
I expect a `bson_impl_inline_t` has an implicit zero `offset` and `depth`.

Does this comment mean a `bson_impl_inline_t` could call `_bson_grow()` and exceed `BSON_MAX_SIZE`?

Reply:
Thank you. The updated comment helped me understand.